### PR TITLE
Jhalackova pxe boot

### PR DESCRIPTION
**Pull request by lvicoun**
### Description

Adding deployment using the PXE client.

### Are there any relevant issues/feature requests?

* bsc#...
* jsc#SLE-...

### Is this (based on) existing content?

* old xml:id...
* old URL...

### CODE REVIEW

**Comment by lvicoun** [comment]
[Micro-deployment-pxe_en.pdf](https://github.com/user-attachments/files/17714872/Micro-deployment-pxe_en.pdf)

File: concepts/deployment-network-boot.xml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    <abstract><!-- can be changed via merge in the assembly -->
+      <para>
+       The topic is an overview of prerequisites needed for network boot.
+      </para>
+    </abstract>
+  </info>
+  
+  <para>
+    This section covers the configuration tasks needed in complex boot scenarios. It contains
+    ready-to-apply configuration examples for DHCP, PXE boot, TFTP, and Wake on LAN. On contrary,
```
Comment (Vogtinator) [inline concepts/deployment-network-boot.xml:33]: Wake on LAN is orthogonal to PXE deployment

File: concepts/configure-combustion.xml [hunk 2]
```diff
@@ -41,7 +41,11 @@
       <command>transactional-update</command> to apply other changes, for
       example, setting &rootuser; password or installing packages.
     </para>
-   
+   <para condition="pxe-boot">
+    The configuration script must reside on the TFTP server and during the boot, its URL must be
+    provided on the kernel CLI using the <literal>combustion.url</literal> parameter. For details,
```
Comment (Vogtinator) [inline concepts/configure-combustion.xml:46]: There's not really a kernel CLI but rather parameters passed to the kernel

File: tasks/deployment-network-dhcp.xml [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+  
+  # The following lines are required
+  option arch code 93 = unsigned integer 16; # RFC4578
+  subnet 192.168.1.0 netmask 255.255.255.0 {
+   next-server 192.168.1.1;
+   range 192.168.1.100 192.168.1.199;
+   default-lease-time 3600;
+   max-lease-time 3600;
+   if option arch = 00:07 or option arch = 00:09 {
+     filename "/EFI/x86/grub.efi";
```
Comment (Vogtinator) [inline tasks/deployment-network-dhcp.xml:76]: Where is this coming from?

I think there's a section missing about the PXE bootloader and bootloader config setup

File: tasks/deployment-network-image.xml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+      </para>
+    </step>
+    <step>
+      <para>
+        Enter UEFI boot menu and select PXE booting.
+      </para>
+    </step>
+    <step>
+      <para>
+        The machine should start the deployment. Kernel and initrd are downloaded from the TFTP.
```
Comment (Vogtinator) [inline tasks/deployment-network-image.xml:48]: This should go into detail ideally with some screenshots about the deployment process

File: tasks/deployment-network-tftp.xml [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+        </para>        
+      </step>
+      <step>
+        <para>
+          Adjust the <filename>SL-Micro.<replaceable>ARCH</replaceable>-&productnumber;.append</filename> file:
+        </para>
+        <itemizedlist>
+          <listitem>
+            <para>
+              <literal>rd.kiwi.install.image.xz</literal> &ndash; provide the full URL to the
```
Comment (Vogtinator) [inline tasks/deployment-network-tftp.xml:128]: The parameter is called `rd.kiwi.install.image`, without `.xz`

File: tasks/deployment-network-tftp.xml [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+              installation image <literal>SL-Micro.<replaceable>ARCH</replaceable>&productnumber;.xz</literal>
```
Comment (Vogtinator) [inline tasks/deployment-network-tftp.xml:129]: The other files like `SL-Micro.*.config.bootoptions`, kernel and initrd need to be in the same directory

File: tasks/deployment-network-tftp.xml [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+            </para>
+          </listitem>
+          <listitem>
+            <para>
+              <literal>rd.kiwi.install.pass.bootparam</literal> &ndash; if additional parameters
```
Comment (Vogtinator) [inline tasks/deployment-network-tftp.xml:134]: Maybe it should be explained that after deployment of the image, kiwi performs kexec into the system. This parameter tells kiwi to pass the kernel parameters from deployment to the first boot of the deployed system.

File: tasks/deployment-network-tftp.xml [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+              needs to be passed to the kernel CLI, for example, the below described ones.
+            </para>
+          </listitem>
+          <listitem>
```
Comment (Vogtinator) [inline tasks/deployment-network-tftp.xml:138]: This should mention `rd.kiwi.oem.installdevice` as well. If this is provided it performs unattended deployment. Without it, a user has to accept a dialog about overwriting all data.

**Review by Vogtinator — CHANGES_REQUESTED** [review]

File: tasks/deployment-network-dhcp.xml (hunk 3 above)
Comment (Vogtinator) [reply tasks/deployment-network-dhcp.xml:76]: There it's handled by the installation of the tftpboot-* package: https://documentation.suse.com/sles/15-SP6/html/SLES-all/cha-deployment-prep-pxe.html#sec-deployment-tftp-server-grub-package

We don't have that on SLM, it's up to the user to "somehow" get the kernel+initrd+cmdline booted on the target host.

**Review by dariavladykina — COMMENTED** [review]
Hi, I've added some comments here. Thanks!

**Pull request merged**